Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1736,56 +1716,25 @@ namespace AZ
             m_cullable.m_isHidden = !isVisible;
         }
 
-        void ModelDataInstance::UpdateMaterialChangeIds()
+        CustomMaterialInfo ModelDataInstance::GetCustomMaterialWithFallback(const CustomMaterialId& id) const
         {
-            // update the material changeId list with the current material assignments
-            m_materialChangeIds.clear();
-
-            for (const auto& materialAssignment : m_materialAssignments)
+            if (auto itr = m_customMaterials.find(id); itr != m_customMaterials.end() && itr->second.m_material)
             {
-                const AZ::Data::Instance<RPI::Material>& materialInstance = materialAssignment.second.m_materialInstance;
-                if (materialInstance.get())
-                {
-                    m_materialChangeIds[materialInstance] = materialInstance->GetCurrentChangeId();
-                }
+                return itr->second;
             }
-        }
 
-        bool ModelDataInstance::CheckForMaterialChanges() const
-        {
-            // check for the same number of materials
-            if (m_materialChangeIds.size() != m_materialAssignments.size())
+            const CustomMaterialId ignoreLodId(DefaultCustomMaterialLodIndex, id.second);
+            if (auto itr = m_customMaterials.find(ignoreLodId); itr != m_customMaterials.end() && itr->second.m_material)
             {
-                return true;
+                return itr->second;
             }
 
-            // check for material changes using the changeId
-            for (const auto& materialAssignment : m_materialAssignments)
+            if (auto itr = m_customMaterials.find(DefaultCustomMaterialId); itr != m_customMaterials.end() && itr->second.m_material)
             {
-                const AZ::Data::Instance<RPI::Material>& materialInstance = materialAssignment.second.m_materialInstance;
-
-                MaterialChangeIdMap::const_iterator it = m_materialChangeIds.find(materialInstance);
-                if (it == m_materialChangeIds.end() || it->second != materialInstance->GetCurrentChangeId())
-                {
-                    return true;
-                }
+                return itr->second;
             }
 
-            return false;
-        }
-
-        void ModelDataInstance::OnRebuildMaterialInstance()
-        {
-            if (m_visible && m_descriptor.m_isRayTracingEnabled)
-            {
-                if (CheckForMaterialChanges())
-                {
-                    m_needsSetRayTracingData = true;
-
-                    // update the material changeId list with the latest materials
-                    UpdateMaterialChangeIds();
-                }
-            }
+            return CustomMaterialInfo{};
```

Comment:
nit: this could be simplified a bit to something like:
```
CustomMaterialInfo ModelDataInstance::GetCustomMaterialWithFallback(const CustomMaterialId& customId) const
{
    const CustomMaterialId ignoreLodId(DefaultCustomMaterialLodIndex, id.second);
    for (auto id : {customId, ignoreLodId, DefaultCustomMaterialId})
    {
        if (auto itr = m_customMaterials.find(id); itr != m_customMaterials.end() && itr->second.m_material)
        {
            return itr->second;
        }
    }
    return CustomMaterialInfo{};
}
```